transportable-udfs-trino-plugin/src/main/java/com/linkedin/transport/trino/TransportConnectorFactory.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.trino;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Module;
+import io.airlift.log.Logger;
+import io.trino.spi.connector.Connector;
+import io.trino.spi.connector.ConnectorContext;
+import io.trino.spi.connector.ConnectorFactory;
+import io.trino.spi.function.FunctionId;
+import java.io.File;
+import java.io.FileFilter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.*;
+
+
+public class TransportConnectorFactory implements ConnectorFactory {
+
+  private static final String DEFAULT_TRANSPORT_UDF_REPO = "/transport-udf-repo";
+  private static final String TRANSPORT_UDF_REPO_CONFIG_NAME = "transport.udf.repo";
+  private static final Logger log = Logger.get(TransportConnectorFactory.class);
+
+  private static final FileFilter TRANSPORT_UDF_JAR_FILTER = (file) -> {
+    return file.isFile() && file.getName().endsWith(".jar") && !file.getName().startsWith("transportable-udfs");
+  };
+
+  private Connector connector;
+  private final Class<? extends Module> module;
+
+
+  // test only
+  public TransportConnectorFactory(Connector connector) {
+    this.connector = connector;
+    this.module = TransportModule.class;
+  }
```

Comment:
can we try to avoid polluting the prod impl with this testing code? we could simply create a test-only factory impl in `TrinoTester` since we aren't actually testing any logic here in that test case:

```java
    ConnectorFactory connectorFactory = new ConnectorFactory() {
      @Override
      public String getName() {
        return "TRANSPORT";
      }
      @Override
      public Connector create(String catalogName, Map<String, String> config, ConnectorContext context) {
        return connector;
      }
    };
```

Reply:
Good point. Done.